src/zero/module/aou.clj

```diff
@@ -174,17 +176,117 @@
   "Use transaction TX to update WORKLOAD statuses."
   [tx workload])
 
+(defn add-aou-workload-table!
+  "Return AoU specific UUID and TABLE for _WORKFLOW-WDL in BODY under transaction TX.
+   This function will only create a new aou workload table if there isn't one, and
+   it uses (analysis_version_number, chip_well_barcode) as the primary key."
+  [tx {:keys [release top] :as _workflow-wdl} body]
+  (let [{:keys [creator cromwell pipeline project]} body
+        {:keys [commit version]} (zero/get-the-version)
+        probe-result (jdbc/query tx ["SELECT * FROM workload WHERE project = ? AND wdl = ? AND release = ?"
+                                     project top release])]
+    (if (seq probe-result)
+      ;; if a table already exists, we return its uuid and the name
+      (let [[{:keys [id uuid]}] probe-result
+            table (format "%s_%09d" pipeline id)]
+        [uuid table])
+      ;; if the expected table doesn't exist, we create record + table and return uuid and name
+      (let [[{:keys [id uuid]}]
+            (jdbc/insert! tx :workload {:commit   commit
+                                        :creator  creator
+                                        :cromwell cromwell
+                                        :input    "aou-inputs-placeholder"
+                                        :output   "aou-outputs-placeholder"
+                                        :project  project
+                                        :release  release
+                                        :uuid     (UUID/randomUUID)
+                                        :version  version
+                                        :wdl      top})
+            table (format "%s_%09d" pipeline id)
+            kind  (format (str/join " " ["UPDATE workload"
+                                         "SET pipeline = '%s'::pipeline"
+                                         "WHERE id = %s"]) pipeline id)
+            work  (format "CREATE TABLE %s OF %s (PRIMARY KEY (analysis_version_number, chip_well_barcode))" table pipeline)]
+        (jdbc/update! tx :workload {:items table} ["id = ?" id])
+        (jdbc/db-do-commands tx [kind work])
+        [uuid table]))))
+
+(comment
```

Comment:
Controversial opinion warning:
I'm not a fan of comment blocks (sorry tom). They cluter the code and quickly go stale.
If this is something that's actually useful for debugging, make it a test.

Reply:
Thanks! I'm not a big fan of comment blocks in production code either, these blocks will go away when this PR's merged. They are here just for the sake of a faster turn around time for myself :octocat: